src/Query/TermsSet.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Elastica\Query;
+
+use Elastica\Exception\InvalidException;
+
+/**
+ * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-set-query.html
+ */
+class TermsSet extends AbstractQuery
+{
+    /**
+     * @param array<bool|float|int|string> $terms
+     */
+    public function __construct(string $field, array $terms)
+    {
+        if ('' === $field) {
+            throw new InvalidException('Terms field name has to be set');
+        }
+
+        if (0 === \count($terms)) {
+            throw new InvalidException('Unable to build Terms query: terms must contains at least one item');
+        }
+
+        $this->setParam($field, ['terms' => $terms]);
+    }
+
+    public function setMinimumShouldMatchField(string $minimumShouldMatchField): self
+    {
+        $params = $this->getParams();
+        $field = \key($params);
+
+        $this->setParam($field, \array_merge($params[$field], ['minimum_should_match_field' => $minimumShouldMatchField]));
+
+        return $this;
+    }
+
+    /**
+     * @param int|string $minimumShouldMatchScript
+     */
+    public function setMinimumShouldMatchScript($minimumShouldMatchScript): self
```

Comment:
We can probably pass an `AbstractScript` here?
Maybe we can pass a string|AbstractScript as well as argument to constructor and call `setMinimumShouldMatchField` when the argument is of type string and `setMinimumShouldMatchScript` when the argument is of type AbstractScript. This way we could avoid to have an invalid partial object until one of these methods is called. This would also avoid to have to check for these params to be set in `toArray`.

@ruflin @thePanz could you chime here?

Reply:
With `AbstractScript` it will be easier to handle more script parameters like `params`, `lang`.
See https://www.elastic.co/guide/en/elasticsearch/painless/current/painless-min-should-match-context.html for a more detailed use of the `minimum_should_match_script` inside `terms_set` query

Reply:
In general I'm in favor of passing in the objects like `AbstractScript` as it means we can reuse functionality. If scripts gets expanded, also this function gets the new features.

Reply:
fixed